protocols/generic_protein_purification/generic_protein_purification.ot2.apiv2.py

```diff
@@ -0,0 +1,613 @@
+                                      multi_transfer=True, **transfer_kwargs):
+        """
+        function description
+
+        :param vol:
+
+        :param index:
+        """
+        nonlocal n_columns, n_samples
+        pip, pip_is_multi = choose_pipette(vol, multi_transfer)
+
+        if pip_is_multi:
+            dest_columns = dest_plate.columns()[0:n_columns]
+            for col in dest_columns:
+                transfer(pip, vol, reag_source_well, col[0],
+                         **transfer_kwargs)
+        else:  # Transfer to each well
+            transfer(pip, vol, reag_source_well,
+                     dest_plate.wells()[0:n_samples],
+                     **transfer_kwargs)
+
+    def remove_supernatant(vol, source_plate, multi_transfer=True,
+                           **transfer_kwargs):
+        """
+        Removes a volume from each sample well of the source plate
+        (well 0 to well n_samples) and transfers it to the liquid trash
+        well of the reservoir.
+
+        :param vol (float): The volume to transfer to the trash.
+
+        :param source_plate (labware plate): The plate to aspirate from.
+
+        :param multi_transfer (Boolean): Use 8-channel pipette and transfer
+        a whole column at a time if true, otherwise use a single channel
+        pipette and transfer well by well.
+
+        :param **transfer_kwargs: Any keyword arguments that you may want
+        to pass into the transfer method calls
+        """
+        nonlocal n_columns, n_samples
+        pip, pip_is_multi = choose_pipette(vol, multi_transfer)
+
+        if pip_is_multi:
+            source_columns = source_plate.columns()[0:n_columns]
+            for col in source_columns:
+                transfer(pip, vol, col[0], waste_well, **transfer_kwargs)
+        else:  # Transfer from each well
+            transfer(pip, vol, source_plate.wells()[0:n_samples],
+                     waste_well, **transfer_kwargs)
+
+    def transfer_plate_to_plate(vol, source_plate, destination_plate):
+        pip, is_multi = choose_pipette(vol, True)
+        if is_multi:
+            for s_col, d_col in zip(source_plate.columns(),
+                                    destination_plate.columns()):
+                transfer(pip, vol, s_col[0], d_col[0])
+        else:
+            transfer(pip, vol, sample_wells, dest_wells)
+
+    # reagents
+
+    '''
+    Define where all reagents are on the deck using the labware defined above.
+
+    e.g.
+
+    water = reservoir12.wells()[-1]
+    waste = reservoir.wells()[0]
+    samples = plate.rows()[0][0]
+    dnase = tuberack.wells_by_name()['A4']
+
+    '''
+
+    lysis_buffer_well = reservoir.wells_by_name()['A1']
+    wash_buffer_well = reservoir.wells_by_name()['A2']
+    elution_buffer_well = reservoir.wells_by_name()['A3']
+    nacl_well = reservoir.wells_by_name()['A4']
+    bead_well = reservoir.wells_by_name()['A5']
+    waste_well = reservoir.wells()[-1]  # Use the last well for waste
```

Comment:
This should be `reservoir.wells()[-1].top(-2)`

Reply:
I changed the code to conform